phoenix-core-server/src/main/java/org/apache/phoenix/replication/reader/ReplicationLogProcessor.java

```diff
@@ -0,0 +1,524 @@
+        // Create a copy of configuration as some of the properties would be
+        // overridden
+        this.conf = HBaseConfiguration.create(conf);
+        this.haGroupName = haGroupName;
+        this.batchSize = this.conf.getInt(REPLICATION_STANDBY_LOG_REPLAY_BATCH_SIZE,
+                DEFAULT_REPLICATION_STANDBY_LOG_REPLAY_BATCH_SIZE);
+        this.batchRetryCount = this.conf.getInt(REPLICATION_STANDBY_BATCH_RETRY_COUNT,
+                DEFAULT_REPLICATION_STANDBY_BATCH_RETRY_COUNT);
+        this.maxRetryDelayMs = this.conf.getLong(REPLICATION_STANDBY_BATCH_RETRY_MAX_DELAY_MS,
+                DEFAULT_REPLICATION_STANDBY_BATCH_RETRY_MAX_DELAY_MS);
+        final int threadPoolSize = this.conf.getInt(REPLICATION_STANDBY_LOG_REPLAY_THREAD_POOL_SIZE,
+                DEFAULT_REPLICATION_STANDBY_LOG_REPLAY_THREAD_POOL_SIZE);
+        decorateConf();
+        this.metrics = createMetricsSource();
+        this.executorService = Executors.newFixedThreadPool(threadPoolSize,
+            new ThreadFactoryBuilder()
+                .setNameFormat("Phoenix-Replication-Log-Processor-" + haGroupName + "-%d")
+                .build());
+    }
+
+    /**
+     * Decorate the Configuration object to make replication more receptive to delays by
+     * reducing the timeout and number of retries.
+     */
+    private void decorateConf() {
+        this.conf.setInt(HConstants.HBASE_CLIENT_RETRIES_NUMBER,
+                this.conf.getInt(REPLICATION_STANDBY_HBASE_CLIENT_RETRIES_COUNT,
+                        DEFAULT_REPLICATION_STANDBY_HBASE_CLIENT_RETRIES_COUNT));
+        this.conf.setLong(HConstants.HBASE_CLIENT_OPERATION_TIMEOUT,
+                this.conf.getLong(REPLICATION_STANDBY_HBASE_CLIENT_OPERATION_TIMEOUT_MS,
+                        DEFAULT_REPLICATION_STANDBY_HBASE_CLIENT_OPERATION_TIMEOUT_MS));
+    }
+
+    public void processLogFile(FileSystem fs, Path filePath) throws IOException {
+
+        // Map from Table Name to List of Mutations
+        Map<TableName, List<Mutation>> tableToMutationsMap = new HashMap<>();
+
+        // Track the total number of processed records from input log file
+        long totalProcessed = 0;
+
+        // Track the current batch size as records will be processed in batch size of
+        // {@link REPLICATION_STANDBY_LOG_REPLAY_BATCH_SIZE}
+        long currentBatchSize = 0;
+
+        LogFileReader logFileReader = null;
+
+        long startTime = System.currentTimeMillis();
+
+        try {
+            // Create the LogFileReader for given path
+            logFileReader = createLogFileReader(fs, filePath);
+
+            for (LogFile.Record record : logFileReader) {
+                final TableName tableName = TableName.valueOf(record.getHBaseTableName());
+                final Mutation mutation = record.getMutation();
+
+                tableToMutationsMap.computeIfAbsent(tableName, k -> new ArrayList<>())
+                        .add(mutation);
+                currentBatchSize++;
```

Comment:
I think we need two batch sizes, one with using the number of mutations and the other using total byte size, that is, batchSize vs batchSizeBytes. For calculating batchSizeBytes, we can use mutation.size().